APE17.rst

```diff
@@ -0,0 +1,701 @@
+using standard packaging tools.
+
+Changes common to all packages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+We first take a look at what changes are needed for all packages, regardless of
+whether they have compiled extensions or not. For pure-Python packages, if all
+these changes are made, astropy-helpers is no longer needed and can be removed
+from the repository.
+
+Running tests
+~~~~~~~~~~~~~
+
+Astropy-helpers provides the ability for anyone to run tests in the source
+distribution to do this with the::
+
+    python setup.py test
+
+command, which behind the scenes built the package, installed it into a
+temporary directory, installed pytest-astropy if needed, and ran the tests
+there.
+
+However, this can now be done more cleanly using `tox`_. Tox is a tool that
+makes it easy to install packages into a temporary clean Python environment
+along with the dependencies declared in setup.cfg, and run a set of custom
+commands - in a sense it does what the test command from astropy-helpers did,
+but in a more general way (including the use of an isolated test environment).
+For example, by putting the following into ``tox.ini``::
+
+    [testenv:test]
+    deps =
+      pytest-astropy
+    commands =
+      pytest aplpy
+
+developers will then easily be able to run the tests with::
+
+    tox -e test
+
+The benefit of using tox over just running pytest directly is that since behind
+the scenes it creates a source distribution of the package and installs it into
+a clean environment, it ensures that source distributions for the package are
+fully functional and that all required files and dependencies are declared. For
+packages with C/Cython extensions it should also guarantee that they are always
+rebuilt before running the tests.
+
+We recognize that one of the benefits of the astropy-helpers infrastructure is
+that developers and users have come to expect that packages in the astropy
+ecosystem all implement ``python setup.py test``, so in order to ensure
+continuity and avoid confusion, since we can't expect user to necessarily
+look at detailed documentation, we recommend that packages that no longer use
+astropy-helpers still make sure that ``python setup.py test`` will not necessarily
+run the tests but at least print out information about how to run the tests, and
+optionally include a link to more information::
+
+    $ python setup.py test
+
+    Note: running tests is no longer done using ``python setup.py test``, but
+    instead you will need to run the following command:
+
+        tox -e test
+
+    If you don't already have tox installed, you can install it with:
+
+        pip install tox
+
+    For more information, see https://docs.astropy.org/...
+
+    $
+
+We could then provide a clear documentation page in a centralized place on how
+to use pytest and how to do common things the test command used to allow (e.g.
+running tests for specific sub-packages, or with remote data).
```

Comment:
I think the parentheses in this sentence should be removed.

Reply:
Done!